### PR TITLE
Entrypoints for pandas >=0.25

### LINKED ISSUE #277: Add pandas.plot entrypoints style

### PR DESCRIPTION
**Pull request by jsignell**
Doesn't yet return proper help ~~or work with `df.plot.scatter()` style calls~~

EDIT: it does work with `df.plot.scatter()` style calls, they just have the same requirements as the df.plot versions (aka plot.scatter requires x and y to be set explicitly whereas hvplot does not).

Closes #277 

Screen Shot 2019-09-25 at 11 39 47 AM
Screen Shot 2019-09-25 at 11 41 01 AM

### CODE REVIEW

**Comment by jsignell** [comment]
I consider this done. There are some outstanding issues to be resolved on the pandas side. But this chunk of work should probably get in sooner rather than later.

**Comment by jsignell** [comment]
Note that in this PR, `by`, `groupby`, and `hover_cols` will only work if the columns that they reference are in `x` or `y`. See https://github.com/pandas-dev/pandas/pull/28647 for the full description.

File: hvplot/plotting/core.py [hunk 1]
```diff
@@ -0,0 +1,679 @@
...
+                param.main.warning("Custom options for existing plot types should not "
+                                   "declare the 'kind' argument. The .%s plot method "
+                                   "was unexpectedly customized with kind=%r."
+                                   % (plot_opts['kind'], name))
+                plot_opts['kind'] = name
+            return hvPlot(self._data, **dict(self._metadata, **plot_opts))
+        return super(hvPlot, self).__getattribute__(name)
+
+
+class hvPlotTab(hvPlot):
```
Comment (philippjfr) [inline hvplot/plotting/core.py]: Renaming this to `hvPlotTabular` would be good.

**Comment by jsignell** [comment]
To Do:
 - [ ] make sure that there are no references to `patch` in the docs

**Comment by jsignell** [comment]
I am fairly confident that the remaining error is unrelated.

**Comment by jsignell** [comment]
I think this is ready to merge

**Pull request merged**